package_test.go

```diff
@@ -1407,3 +1373,22 @@ func (s *PackageSuite) TestRaceConditionFinalizerTX(c *C) {
 	// Assert that sql.Stmt was not closed early.
 	c.Assert(q.Run(), IsNil)
 }
+
+func (s *PackageSuite) TestStatementTXReuse(c *C) {
```

Comment:
My reasoning was to have a different test so we do not accidentally remove it in a refactor or conflate it with something different in the general transaction tests. I am happy to move it if you think it will not hinder readability nor future refactors.